### PR TITLE
Add EmailApi

### PR DESCRIPTION
**Pull request by OleksiiMaksymov**
MODLOGIN-60
 * Initial module setup
 * Add the endpoint /email with POST method
 * Implement EmailAPI and MailServiceImpl services
 * Add tests

### CODE REVIEW

File: descriptors/DeploymentDescriptor-template.json [hunk 1]
```diff
@@ -0,0 +1,4 @@
+{
+  "srvcId": "${artifactId}-${version}",
+  "nodeId": "10.0.2.15"
```
Comment (OleksiiKuzminov) [inline descriptors/DeploymentDescriptor-template.json]: add plz run command like this for local or vagrant run
 "descriptor": {
    "exec": "java -Dport=%p -jar ../mod-email/target/mod-email-fat.jar -Dhttp.port=%p embed_postgres=true"
  }

Comment (OleksiiKuzminov) [inline descriptors/DeploymentDescriptor-template.json]: "nodeId": "localhost"

File: descriptors/ModuleDescriptor-template.json [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+  "provides": [
+    {
+      "id": "email",
+      "version": "1.0",
+      "handlers": [
+        {
+          "methods": [
+            "POST"
+          ],
+          "pathPattern": "/email"
```
Comment (OleksiiKuzminov) [inline descriptors/ModuleDescriptor-template.json]: plz add standard permission for this endpoint

File: src/main/java/org/folio/services/MailService.java [hunk 3]
```diff
@@ -0,0 +1,10 @@
+package org.folio.services;
+
+import io.vertx.core.Future;
+import io.vertx.core.json.JsonObject;
+import org.folio.rest.jaxrs.model.Configurations;
+import org.folio.rest.jaxrs.model.EmailEntity;
+
+public interface MailService {
```
Comment (OleksiiKuzminov) [inline src/main/java/org/folio/services/MailService.java]: plz, use vertx event bus for async services

Comment (OleksiiKuzminov) [reply src/main/java/org/folio/services/MailService.java]: all interface methods must have javadocs

Comment (OleksiiMaksymov) [reply src/main/java/org/folio/services/MailService.java]: thanks.
made changes to the code.

File: descriptors/ModuleDescriptor-template.json [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+      "id": "email",
+      "version": "1.0",
+      "handlers": [
+        {
+          "methods": [
+            "POST"
+          ],
+          "pathPattern": "/email",
+          "permissionsRequired": [
+            "configuration.entries.item.get"
```
Comment (OleksiiKuzminov) [inline descriptors/ModuleDescriptor-template.json]: Wrong permission name or it should be modulePermission?

File: descriptors/ModuleDescriptor-template.json [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+          ],
+          "pathPattern": "/_/tenant"
+        }
+      ]
+    }
+  ],
+  "requires": [],
+  "permissionSets": [
+    {
+      "permissionName": "configuration.entries.collection.get",
```
Comment (OleksiiKuzminov) [inline descriptors/ModuleDescriptor-template.json]: Permissions from another module?

**Comment by OleksiiMaksymov** [comment]
Thanks for the comments.
Done

**Pull request merged**